packages/block-library/src/media-text/save.js

```diff
@@ -33,9 +33,34 @@ export default function save( { attributes } ) {
 		verticalAlignment,
 		imageFill,
 		focalPoint,
+		linkClass,
+		href,
+		linkTarget,
+		rel,
 	} = attributes;
+	const newRel = isEmpty( rel ) ? undefined : rel;
+
+	let image = <img
```

Comment:
I guess we could assign image to a function, so the element is not computed if the media type is not an image.
```suggestion
	let image;
	if ( href ) {
		image = () => (
			<a
				className={ linkClass }
				href={ href }
				target={ linkTarget }
				rel={ newRel }
			>
				{ image }
			</a>
		);
	} else {
		image = () => (
			<img
				src={ mediaUrl }
				alt={ mediaAlt }
				className={ ( mediaId && mediaType === 'image' ) ? `wp-image-${ mediaId }` : null }
			/>
		);
	}
```